src/main/java/dev/blachut/svelte/lang/parsing/js/SvelteJSParser.kt

```diff
@@ -22,4 +25,16 @@ class SvelteJSParser(builder: PsiBuilder) : ES6Parser<ES6ExpressionParser<*>, ES
             }
         }
     }
+
+    override fun buildTokenElement(type: IElementType) {
+        // there are too many places that uses element type JSElementTypes.REFERENCE_EXPRESSION,
+        // so use the new one only for the specific references
+        return super.buildTokenElement(
```

Comment:
Interesting, I tried to use `buildTokenElement` but had some problems with it, because not every `marker.done` goes through this method. 

I found one example that does not work that way:

https://svelte.dev/repl/65fc4b475b884dcba414139848ff02ef?version=3.29.0
```
const count = writable(0);
const x = { $count };
console.log(x); // Prints { $count: 0 }
```

Anyway this looks like weird side effect of stores implementation and not a feature so I guess it's okay to ignore it for now.

Reply:
yes, I think so. Unfortunately, we cannot get the token text using the other implementation because it is called after "advance()" method